src/core/lib/gprpp/thd.h

```diff
@@ -92,14 +92,14 @@ class Thread {
   /// Normal constructor to create a thread with name \a thd_name,
   /// which will execute a thread based on function \a thd_body
   /// with argument \a arg once it is started.
-  /// The optional \a success argument indicates whether the thread
+  /// The \a success argument indicates whether the thread
   /// is successfully created.
   /// The optional \a options can be used to set the thread detachable.
   Thread(const char* thd_name, void (*thd_body)(void* arg), void* arg,
-         bool* success = nullptr, const Options& options = Options());
+         bool* success, const Options& options = Options());
```

Comment:
We still need old way of creating Thread because internal code (that you cannot access) still need this.

Reply:
Adding a new constructor that does not require a `success` argument would undermine the point of this change. It would allow future code to use this constructor and thereby re-introduce the problem we have that pthread_create might fail and the result is a silent, slow degradation of functionality which is very difficult to diagnose.

Reply:
It'd be true but we cannot accept it without having a way to not break internal codes.